run.py

```diff
@@ -1,8 +1,38 @@
 from environment import Environment
-
+from enums import EntityTypes, ContainerState
 
 environment = Environment()
 environment.setup()
 
 for _ in range(environment.config.run_length):  # run model!
-    pass
+
+    for producer in environment.producers:
+        producer.produce() # all producers produce according to their production rate
+        for shipment in producer.storage: # all shipments that are produced are registered at the auction
+            if EntityTypes.__members__['SHIPMENT'] not in \
+                        producer.region.auctioneer.entities.keys(): # the first if statement would not be necessary if the dict already has the key...
+                registrationkey = producer.region.auctioneer.register(shipment)
+                bid = producer.bid(registrationkey,shipment)
+                producer.region.auctioneer.list_shipment(bid)
+            elif shipment not in producer.region.auctioneer.entities[EntityTypes.SHIPMENT].values():
+                registrationkey = producer.region.auctioneer.register(shipment)
+                bid = producer.bid(registrationkey, shipment)
+                producer.region.auctioneer.list_shipment(bid)
```

Comment:
Why are you doing this twice?

Reply:
the first if statement is required when the key EntityTypes.SHIPMENT is not added to the dictionary, in otherwords, the case when the first shipment needs to be registered.

Reply:
I meant why are you repeating yourself: just put both into one if statement (an `or` should do what you want).